src/Features/Core/Portable/ExtractMethod/MethodExtractor.CodeGenerator.cs

```diff
@@ -83,97 +80,93 @@ protected CodeGenerator(InsertionPoint insertionPoint, SelectionResult selection
+                // the original type was nullable. This allows our code generation to follow our recommendation of only
+                // using nullable when necessary. This is done after method generation instead of at analyzer time
+                // because it's purely based on the resulting code, which the generator can modify as needed. If return
+                // statements are added, the flow analysis could change to indicate something different. It's cleaner to
+                // rely on flow analysis of the final resulting code than to try and predict from the analyzer what will
+                // happen in the generator. 
+                var finalDocument = await UpdateMethodAfterGenerationAsync(callSiteDocument, newMethodDefinition, cancellationToken).ConfigureAwait(false);
+
+                return await CreateGeneratedCodeAsync(newMethodDefinition.Status, finalDocument, cancellationToken).ConfigureAwait(false);
+            }
 
-                var newCallSiteRoot = callSiteDocument.Root;
+            private async Task<SemanticDocument> InsertMethodAndUpdateCallSiteAsync(
+                IMethodSymbol newMethodDefinition, CancellationToken cancellationToken)
+            {
+                var document = this.SemanticDocument.Document;
+                var codeGenerationService = document.GetLanguageService<ICodeGenerationService>();
 
-                var codeGenerationService = SemanticDocument.Document.GetLanguageService<ICodeGenerationService>();
-                var result = GenerateMethodDefinition(LocalFunction, cancellationToken);
+                // First, update the callsite with the call to the new method.
+                var outermostCallSiteContainer = GetOutermostCallSiteContainerToProcess(cancellationToken);
 
-                SyntaxNode destination, newContainer;
-                if (LocalFunction)
-                {
-                    destination = InsertionPoint.With(callSiteDocument).GetContext();
+                var rootWithUpdatedCallSite = this.SemanticDocument.Root.ReplaceNode(
+                    outermostCallSiteContainer,
+                    await GenerateBodyForCallSiteContainerAsync(outermostCallSiteContainer, cancellationToken).ConfigureAwait(false));
 
-                    // No valid location to insert the new method call.
-                    if (destination == null)
-                    {
-                        return await CreateGeneratedCodeAsync(
-                            OperationStatus.NoValidLocationToInsertMethodCall, callSiteDocument, cancellationToken).ConfigureAwait(false);
-                    }
+                // Then insert the local-function/method into the updated document that contains the updated callsite.
+                var documentWithUpdatedCallSite = await this.SemanticDocument.WithSyntaxRootAsync(rootWithUpdatedCallSite, cancellationToken).ConfigureAwait(false);
```

Comment:
the top-part above this is teh same for both local functions and methods.  we take the starting selection and replace it with a call to teh new thing we're going to generate.  

after that, we have customized logic for either inserting a locl function or inserting a method.